src/main/java/randoop/generation/ComponentManager.java

```diff
@@ -60,6 +65,15 @@ public class ComponentManager {
    */
   private final Collection<Sequence> gralSeeds;
 
+  /** For each scope in the SUT, statistics about its constants. */
+  public ScopeToConstantStatistics scopeToConstantStatistics = new ScopeToConstantStatistics();
+
+  /**
+   * Cache for constant sequences filtered by scope and type. Key format: scopeKey + ":" +
+   * neededType + ":" + onlyReceivers
+   */
+  private final Map<String, SIList<Sequence>> constantSequenceCache = new HashMap<>();
+
   /**
    * Components representing literals that should only be used as input to specific classes.
```

Comment:
Are the fields `classLiterals` and `packageLiterals` needed?  I think that uses of them can instead use `scopeToConstantStatistics`.  Then we can eliminate three classes from Randoop:  `ClassLiterals`, `PackageLiterals`, and `MappedSequences`.